ui/platform_window/x11/x11_window_base.cc

```diff
@@ -273,11 +274,17 @@ void X11WindowBase::SetCapture() {}
 
 void X11WindowBase::ReleaseCapture() {}
 
-void X11WindowBase::ToggleFullscreen() {}
+void X11WindowBase::ToggleFullscreen() {
+  SetWMSpecState(!is_fullscreen_,
+                 atom_cache_.GetAtom("_NET_WM_STATE_FULLSCREEN"), None);
+  is_fullscreen_ = !is_fullscreen_;
+}
 
 void X11WindowBase::Maximize() {
-  if (IsMaximized())
-    return;
+  // Unfullscreen the window if it is fullscreen.
+  if (HasWMSpecProperty("_NET_WM_STATE_FULLSCREEN"))
```

Comment:
can we use IsFullscreen within the 'if' clause?

maybe add a DCHECK in IsFullscreen so that is_fullscreen_ matches the result of  (HasWMSpecProperty("_NET_WM_STATE_FULLSCREEN"))?